cirq/optimizers/merge_single_qubit_gates_test.py

```diff
@@ -33,8 +33,8 @@ def assert_optimizes(
         cirq.DropEmptyMoments()
     ]
     for post in followup_optimizations:
-        post.optimize_circuit(before)
-        post.optimize_circuit(expected)
+        post.optimize_circuit(before) # type: ignore
```

Comment:
Not ignoring types gives me: 
```
cirq/optimizers/merge_single_qubit_gates_test.py:36: error: "object" has no attribute "optimize_circuit"
cirq/optimizers/merge_single_qubit_gates_test.py:37: error: "object" has no attribute "optimize_circuit"
```
How to fix?

Reply:
All uses of `optimize_circuit` should be changed to direct calls, i.e. `FooOptPass.optimize_circuit(circuit)` becomes just `FooOptPass(circuit)`. 

For all existing subclasses of `OptimizationPass`, the `optimize_circuit` method should be renamed to `__call__` (and, obviously, they should no longer subclass `OptimizationPass`).

Wherever an object of type `OptimizationPass` is specified, the type should be changed to `Callable[[Circuit], None]`.

Reply:
With the update
```
cirq/optimizers/merge_single_qubit_gates_test.py:36: error: "object" not callable
cirq/optimizers/merge_single_qubit_gates_test.py:37: error: "object" not callable
```